mycpp/smartptr.h

```diff
@@ -2,8 +2,6 @@
 
 #include <assert.h>
 
-#include "mycpp/gc_heap.h"
```

Comment:
Please revert this, since when `clang-format` sorts the #includes it causes a build error

The smartptr.h file uses gHeap, which is in gc_heap.h, so it should be included

I tried reverting the style change to the test only, and it caused a build error, so both files should be reverted

I think smartptr* has nothing to do with this change anyway -- it's unused in the codebase.


clang-format is basically enforcing this style:

https://github.com/include-what-you-use/include-what-you-use